src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php

```diff
@@ -38,14 +38,25 @@ public function publish(Message $message, $key = null)
 
             unset($properties['application_headers']);
         }
-
+        
         $this->exchange->publish(
             $message->getBody(),
             $key,
             $this->flags,
-            $properties
+            $this->removeArraysFromHeaders($properties)
         );
     }
+    
+    private function removeArraysFromHeaders(array $properties)
```

Comment:
Can you:
* rename this method with something like `sanitizeProperties`?
* move the code regarding `delivery_mode` from the RetryProcessor to this method?

Thanks. :)